go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -649,6 +661,19 @@ func (srv *mysqlServer) shutdownMysqlProtocolAndDrain() {
 	}
 }
 
+// stopListener Close or Shutdown a mysql listener depending on the shutdown argument.
+func stopListener(listener *mysql.Listener, shutdown bool) {
+	if listener == nil {
+		return
+	}
+	if shutdown {
+		listener.Shutdown()
+	} else {
+		listener.Close()
+	}
+	listener = nil
```

Comment:
this is passed as reference, setting this to nil would not change the pointer reference of `srv.unixListener` or `srv.tcpListener`

Reply:
Fixed via f1b0d53281